File: problems/1209/solution.go
package main
func removeDuplicates(s string, k int) string {
    ra, i := []rune(s), 0
    
    for j := 0; j < len(ra); i, j = i + 1, j + 1 {
        ra[i] = ra[j]
        
        if (i >= k - 1 && check(ra, i, k)) {
            i -= k
        }
    }
    
    return string(ra[:i])
}

func check(ra []rune, i, k int) bool {
    for j := i - k + 1; j < i; j++ {
        if ra[i] != ra[j] {
            return false
        }
    }
    
    return true
}

File: problems/1209/solution.cpp
class Solution {
public:
    string removeDuplicates(string s, int k) {
        int j = 0;
        stack<int> st;
        
        for (int i = 0; i < size(s); i++, j++) {
            s[j] = s[i];
            
            if (j == 0 or s[j] != s[j - 1]) {
                st.push(1);
            } else if (++st.top() == k) {
                j -= k;
                st.pop();
            }
        }
        
        return s.substr(0, j);
    }
};